### ISSUE TITLE
`require-meta-schema` should allow variable schema

### ISSUE THREAD

**Issue by bmish** [not-an-aardvark/eslint-plugin-eslint-plugin#273]
If someone wants to use a variable or function call to get the value for any of the meta properties, that should be allowed.

```js
{
  schema: baseRule.meta.schema
}
```

```js
{
  schema: getSchema()
}
```

Example: https://github.com/typescript-eslint/typescript-eslint/blob/ad412cdd74dc5619fbe4bf27c0a5eb5c5a4b65ca/packages/eslint-plugin/src/rules/brace-style.ts#L27

This issue is not specific to TypeScript.
